ENIGMAsystem/SHELL/Universal_System/roomsystem.cpp

```diff
@@ -268,6 +245,32 @@ namespace enigma
 
     instance_event_iterator = new inst_iter(NULL,NULL,NULL);
 
+    // Initialize view variants so they do not throw uninitialized variable access errors.
+    view_enabled = views_enabled;
+    for (unsigned i=0;i<8;i++) {
+      view_xview[i] = views[i].area_x; view_yview[i] = views[i].area_y; view_wview[i] = views[i].area_w; view_hview[i] = views[i].area_h;
+      view_xport[i] = views[i].port_x; view_yport[i] = views[i].port_y; view_wport[i] = views[i].port_w; view_hport[i] = views[i].port_h;
+      view_object[i] = views[i].object2follow;
+      view_hborder[i] = views[i].hborder; view_vborder[i] = views[i].vborder; view_hspeed[i] = views[i].hspd; view_vspeed[i] = views[i].vspd;
+      view_visible[i] = (bool)views[i].start_vis;
+      view_angle[i] = 0;
+      // do the object follow once to initialize the view correctly
+      int vob = (int)view_object[i];
+      if (vob != -1)
```

Comment:
This check isn't doing much; `vob` is equally likely to be `noone`.

Reply:
We should check `object_exists` and `instance_exists` actually since this can also be an instance id.

Reply:
We decided to leave it alone over Discord to support allowing it to be `-1` since `follow_object` itself already checks `instance_exists`.

Reply:
Yes. It's unclear how a check for `self` will behave outside of an instance; it might be worth testing this, though, because this check should theoretically still  be redundant.

Reply:
I think the check should honestly be `vob >= 0` because of what I just found in the GMSv1.4 manual:
https://docs.yoyogames.com/source/dadiospice/002_reference/windows%20and%20views/views/view_object.html

> To prevent any view following, you can set this variable using the special keyword noone.